### PR TITLE
tests: bluetooth: tester: allow to add more CCCs

### PR DESCRIPTION
**Pull request by KKopyscinski**
Specification states that: "Only one Client Characteristic Configuration declaration shall exist in a characteristic definition." This was wrongly understood as "Ony one CCC can exist in GATT Server". When we add next characteristic `ccc_added` flag is cleared so next CC can be added (for this characteristic).

Signed-off-by: Krzysztof Kopyściński <krzysztof.kopyscinski@codecoup.pl>

### CODE REVIEW

**Review by hermabe — DISMISSED** [review]

File: tests/bluetooth/tester/src/btp_gatt.c [hunk 1]
```diff
@@ -483,8 +487,6 @@ static void add_characteristic(uint8_t *data, uint16_t len)
 		   CONTROLLER_INDEX, BTP_STATUS_FAILED);
 }
 
-static bool ccc_added;
-
 static uint8_t ccc_value;
```
Comment (sjanc) [inline tests/bluetooth/tester/src/btp_gatt.c:490]: ccc_value is shared ,  so I think those (ccc_value and ccc_added) should be per characteristic (maybe in user data 'struct gatt_value'?)

**Review by sjanc — CHANGES_REQUESTED** [review]

File: tests/bluetooth/tester/src/btp_gatt.c (hunk 1 above)
Comment (KKopyscinski) [reply tests/bluetooth/tester/src/btp_gatt.c:490]: yes, at least `ccc_value` (we add these descriptors and set characteristic values in sequence, so most recently added characteristic is updated). And hence `user_data` of CCC is not used I think it's good place to add this

**Comment by KKopyscinski** [comment]
@sjanc I think it's ok now

**Review by sjanc — DISMISSED** [review]

**Comment by sjanc** [comment]
#AutoPTS run zephyr GATT/SR

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/zephyrproject-rtos/zephyr/pull/52816#issuecomment-1340855514 after 12:53:10.

**Comment by sjanc** [comment]
#AutoPTS run zephyr GATT/SR

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/zephyrproject-rtos/zephyr/pull/52816#issuecomment-1342225700 after 08:57:58.

**Comment by sjanc** [comment]
#AutoPTS run zephyr GATT/SR

**Comment by codecoup-tester** [comment]
Scheduled PR #https://github.com/zephyrproject-rtos/zephyr/pull/52816#issuecomment-1342621661 after 13:03:29.

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
Failed tests:
GATT    GATT/SR/GAN/BV-02-C             BTP ERROR
GATT    GATT/SR/GAR/BI-11-C             UNKNOWN VERDICT: NONE
GATT    GATT/SR/GAW/BV-14-C             BTP ERROR
GATT    GATT/SR/GPM/BV-05-C             INCONC

**Review by sjanc — APPROVED** [review]

File: tests/bluetooth/tester/src/btp_gatt.c [hunk 2]
```diff
@@ -788,6 +833,15 @@ static uint8_t alloc_value(struct bt_gatt_attr *attr, struct set_value *data)
 
 	memcpy(value->data, data->value, value->len);
 
+	/** Handle of attribute is 1 less that handle to its value */
```
Comment (hermabe) [inline tests/bluetooth/tester/src/btp_gatt.c:836]: This is not true in general, there may be other descriptors before the CCCD, but I think it is ok for these tests. No change requested.

**Review by hermabe — APPROVED** [review]

**Pull request merged**